### PR TITLE
Refresh Documentation

### LINKED ISSUE #131: Modernize Docs for Version 1.1.0
The docs need a thorough review and update to bring them up to the latest version of Gaphor.

- Technical editing improvements
- New screenshots and updated user manual
- Updated installation and development instructions

### I'm submitting a...
- [ ] Bug report  
- [ ] Feature request
- [X] Documentation issue or request

### PR DESCRIPTION
**Pull request by danyeaw**
Closes #131.

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and am following the [Contributer guide](https://github.com/gaphor/gaphor/blob/master/CONTRIBUTING.md)
- [X] I have read and understand the [Code of Conduct](https://github.com/gaphor/gaphor/blob/master/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Documentation content changes

### What is the current behavior?

Issue Number: N/A

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: docs/model.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+The Python data model is generated from a Gaphor model file that describes
+the relationships between the supported UML elements.
 
-The model is built using smart properties (descriptors). Those
-descriptors handle events when they're changed. This allows the rest of
-the application (visuals, undo system) to update their state
-accordingly. The events are send using Zope (3)'s signalling mechanism,
-called Handlers.
+The model is built using smart properties (descriptors). These properties
+handle events when they're changed. This allows the rest of the application,
```
Comment (amolenaar) [inline docs/model.md]: `handle events` should be `emit events` (?)

File: docs/multidispatching.md [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
 
-    from generic.multidispatching import multidispatch
+    from gaphor.utils.generic.multidispatching import multidispatch
 
     @multidispatch(Dog)
     def sound(o):
-      print "Woof!"
+      print("Woof!")
 
     @sound.when(Cat)
```
Comment (amolenaar) [inline docs/multidispatching.md]: `when` should be `register`, as is with single-dispatch.

File: docs/services.md [hunk 3]
```diff
@@ -18,25 +15,22 @@ service itself and therefore implements Service too.
...
-stores the add/remove signals in the undo system.
-
-The same goes for UML.Elements. Those classes (or more specific the
-properties) send notifications every time their state changes.
+A nice example of a service in use is the ElementFactory. Currently it is
+tightly bound to the `gaphor.UML` module. A default factory is created in
+`__init__.py`.
 
-But.. where to put such information?
+The ElementFactory depends on the undo_manager. When an important events
```
Comment (amolenaar) [inline docs/services.md]: doesn't the undo manager depend on events emitted by the element factory?

File: docs/so.md [hunk 4]
```diff
@@ -1,58 +1,46 @@
...
 
 ```eval_rst
 .. autoclass:: gaphor.abc.Service
 ```
 
-UI components is another piece of cake. The `gaphor.uicomponents` entry
-point group is used to define windows and user interface functionality.
-A UI component should implement the gaphor.ui.abc.UIComponent interface:
+Another more specialized service that also inherits from `gaphor.abc.Service`,
+is the UI Components service. Services that use this interface are used to
```
Comment (amolenaar) [inline docs/so.md]: `UI Component service` (without the s)

File: docs/framework.md [hunk 5]
```diff
@@ -1,27 +1,39 @@
...
-
-Gaphor is transactional. Transactions work simply by sending an event
-when a transaction starts and sending another when a transaction ends.
-E.g. undo management is transactional.
-
-It all starts with an Application. Only one Application instance is
-permitted. The Application will look for services defined as
-[gaphor.services](services.md). Those services are loaded and initialized.
+split in a series of services, such as a file manager, undo manager and GUI
+manager. Those services are loaded based on entry points defined in the
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply docs/framework.md]: Good catch. I'll use the event manager as another example.

File: docs/multidispatching.md (hunk 2 above)
Comment (danyeaw) [reply docs/multidispatching.md]: I think we can remove this page now that we have are just using the generic module. This probably needs to be updated in the generic docs.

**Pull request merged**